tests/test_client.py

```diff
@@ -295,6 +294,180 @@ def test_sync_score_ok(self):
+            mock_post.assert_called_with(
+                'https://api3.siftscience.com/v3/accounts/ACCT/users/%s/decisions' % user_id,
+                auth=mock.ANY, data=data, headers=mock.ANY, timeout=mock.ANY)
+        except Exception as e:
+            assert(isinstance(e, sift.client.ApiException))
+
+    def test_apply_decision_invalid_source_fails(self):
+        user_id = '54321'
+        mock_response = mock.Mock()
+        applyDecisionRequest = {
+            'decision_id': 'user_looks_ok_legacy',
+            'source': 'INVALID_SOURCE',
+            'time': 1481569575
+        }
+        with mock.patch('requests.post') as mock_post:
+            mock_post.return_value = mock_response
+        try:
+            response = self.sift_client.apply_user_decision(user_id, applyDecisionRequest)
+            data = json.dumps(applyDecisionRequest)
+            mock_post.assert_called_with(
+                'https://api3.siftscience.com/v3/accounts/ACCT/users/%s/decisions' % user_id,
+                auth=mock.ANY, data=data, headers=mock.ANY, timeout=mock.ANY)
+        except Exception as e:
+            assert(isinstance(e, sift.client.ApiException))
+
+    def test_apply_decision_to_order_ok(self):
+        user_id = '54321'
+        order_id = '43210'
+        mock_response = mock.Mock()
+        applyDecisionRequest = {
+                'decision_id': 'order_looks_bad_payment_abuse',
+                'source': 'AUTOMATED_RULE',
+                'time': 1481569575
+            }
+
+        applyDecisionResponseJson = '{' \
+                                    '"entity": {' \
+                                        '"id": "54321",' \
+                                        '"type": "order"'    \
+                                    '},' \
+                                    '"decision": {' \
+                                        '"id":"order_looks_bad_payment_abuse"' \
+                                    '},' \
+                                    '"time":"1481569575"}'
+
+        mock_response.content = applyDecisionResponseJson
+        mock_response.json.return_value = json.loads(mock_response.content)
+        mock_response.status_code = 200
+        mock_response.headers = response_with_data_header()
+        with mock.patch('requests.post') as mock_post:
+            mock_post.return_value = mock_response
+            response = self.sift_client.apply_order_decision(user_id, order_id, applyDecisionRequest)
+            data = json.dumps(applyDecisionRequest)
+            mock_post.assert_called_with(
+                'https://api3.siftscience.com/v3/accounts/ACCT/users/%s/orders/%s/decisions' % (user_id,order_id),
+                auth=mock.ANY, data=data, headers=mock.ANY, timeout=mock.ANY)
+            assert(isinstance(response, sift.client.Response))
+            assert(response.is_ok())
+            assert(response.http_status_code == 200)
+            assert(response.body['entity']['type'] == 'order')
+
```

Comment:
Can we add some testing for the logic in `validate_apply_decision_request()` please?